### PR TITLE
5432/pubsub discover

### PR DESCRIPTION
**Pull request by sblack-usu**
This PR requires service account credentials to be placed at `/service-account-pubsub.json`. By going straight to pubsub rather than using mongo changestream, listeners can be serverless and run on cloud run.  I will setup a cloud run job that subscribes to the `discovery_ids` queue, collects the metadata and places/removes the metadata in the mongo discovery collection

The pubsub library is asynchronous so running the job shouldn't affect performance in a significant way.  Each time the signal processor is called, the resource id is packaged into json, converted to bytes and sent asynchronously.

### CODE REVIEW

**Review by devincowan — DISMISSED** [review]
wow, that's nice and simple!

File: hs_core/pubsub_discovery_processor.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from django.conf import settings
+from google.auth import jwt
+from google.cloud import pubsub_v1
+
+publish_discoverable = getattr(settings, "PUBLISH_DISCOVERABLE", False)
+
+if publish_discoverable:
+    audience = "https://pubsub.googleapis.com/google.pubsub.v1.Publisher"
+    credentials = jwt.Credentials.from_service_account_file('service-account-pubsub.json', audience=audience)
```
[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_core/pubsub_discovery_processor.py:11]: Good idea

[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_core/pubsub_discovery_processor.py:11]: Added in hydroshare/settings.py.

**Review by devincowan — APPROVED** [review]

**Review by devincowan — APPROVED** [review]

**Comment by devincowan** [comment]
resolves https://github.com/hydroshare/hydroshare/issues/5432

**Pull request merged**